validator/src/journal/block_manager.rs

```diff
@@ -398,13 +402,200 @@ impl BlockManagerState {
+        block_id: &str,
+        ids: &[&String],
+    ) -> Result<bool, BlockManagerError> {
+        let _lock = self
+            .persist_lock
+            .read()
+            .expect("The persist RwLock is poisoned");
+        let blockstore_by_name = self
+            .state
+            .blockstore_by_name
+            .read()
+            .expect("The blockstore RwLock is poisoned");
+        if let Some(store) = self.persisted_branch_contains_block(&blockstore_by_name, block_id)? {
+            self.persisted_branch_contains_any_transactions(store, block_id, ids)
+        } else {
+            if block_id != NULL_BLOCK_IDENTIFIER {
+                for pool_block in self.branch(block_id)? {
+                    if let Some(store) = self.persisted_branch_contains_block(
+                        &blockstore_by_name,
+                        &pool_block.header_signature,
+                    )? {
+                        return self.persisted_branch_contains_any_transactions(
+                            store,
+                            &pool_block.header_signature,
+                            ids,
+                        );
+                    }
+                    if self.block_contains_any_transactions(&pool_block, ids) {
+                        return Ok(true);
+                    }
+                }
+            }
+
+            Ok(false)
+        }
+    }
+
+    pub fn contains_any_batches(
+        &self,
+        block_id: &str,
+        ids: &[&String],
+    ) -> Result<bool, BlockManagerError> {
+        let _lock = self
+            .persist_lock
+            .read()
+            .expect("The persist RwLock is poisoned");
+        let blockstore_by_name = self
+            .state
+            .blockstore_by_name
+            .read()
+            .expect("The blockstore RwLock is poisoned");
+        if let Some(store) = self.persisted_branch_contains_block(&blockstore_by_name, block_id)? {
+            self.persisted_branch_contains_any_batches(store, block_id, ids)
+        } else {
+            if block_id != NULL_BLOCK_IDENTIFIER {
+                for pool_block in self.branch(block_id)? {
+                    if let Some(store) = self.persisted_branch_contains_block(
+                        &blockstore_by_name,
+                        &pool_block.header_signature,
+                    )? {
+                        return self.persisted_branch_contains_any_batches(
+                            store,
+                            &pool_block.header_signature,
+                            ids,
+                        );
+                    }
+                    if self.block_contains_any_batches(&pool_block, ids) {
+                        return Ok(true);
+                    }
+                }
+            }
+            Ok(false)
+        }
+    }
+
+    fn block_contains_any_transactions(&self, block: &Block, ids: &[&String]) -> bool {
+        let block_ids: HashSet<&String> = HashSet::from_iter(
```

Comment:
Is transaction_ids a better name for this?

Reply:
Yes, thanks!.